android/uhabits-android/src/main/java/org/isoron/uhabits/widgets/NumericalCheckmarkWidget.kt

```diff
@@ -0,0 +1,49 @@
+package org.isoron.uhabits.widgets
+
+import android.app.PendingIntent
+import android.content.Context
+import android.content.Intent
+import android.util.Log
+import android.view.View
+import org.isoron.uhabits.core.models.Checkmark
+import org.isoron.uhabits.core.models.Habit
+import org.isoron.uhabits.receivers.WidgetReceiver
+import org.isoron.uhabits.receivers.WidgetReceiver.Companion.ACTION_SET_NUMERICAL_VALUE
+import org.isoron.uhabits.utils.PaletteUtils
+import org.isoron.uhabits.widgets.activities.NumericalCheckmarkWidgetActivity
+import org.isoron.uhabits.widgets.views.CheckmarkWidgetView
+import org.isoron.uhabits.widgets.views.NumericalCheckmarkWidgetView
+
+class NumericalCheckmarkWidget(context: Context, widgetId: Int, habit: Habit) : CheckmarkWidget(context, widgetId, habit) {
```

Comment:
Instead of creating two new classes (`NumericalCheckmarkWidget` and `NumericalCheckmarkWidgetView`) which inherit from the existing classes (`CheckmarkWidget` and `CheckmarkWidgetView`), I would prefer if you just modified the two existing classes instead. Currently, you are essentially using inheritance to avoid some conditionals in the methods `CheckmarkWidget.getOnClickPendingIntent` and `CheckmarkWidgetView.getText`. This makes the code less clear and creates classes which contain essentially boilerplate code. Please remove the classes `NumericalCheckmarkWidget`, `NumericalCheckmarkWidgetView` and use conditionals instead.